src/portkey/awssig.clj

```diff
@@ -126,3 +127,75 @@
         headers (assoc headers "Authorization" auth)]
     (assoc req :headers headers)))
 
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Signing a request v2
+;; https://docs.aws.amazon.com/AmazonSimpleDB/latest/DeveloperGuide/HMACAuth.html#REST_RESTAuth
+
+(def x-amz-date-formatter-sign-2
+  (java.time.format.DateTimeFormatter/ofPattern "yyyy-MM-dd'T'HH:mm:ss'Z'"))
+
+(defn- hmac-sha-256-sign2
+   [^bytes secret ^String s]
+   (-> (javax.crypto.Mac/getInstance "HmacSHA256")
+         (doto (.init (javax.crypto.spec.SecretKeySpec. secret "HmacSHA256")))
+         (.doFinal (.getBytes s "UTF8"))))
+
+(defn- utf8-natural-byte-cmp
```

Comment:
This requirement felt bizarre initially but now I understand it: it's the simple unambiguous order one can implement over strings. Default orders are often dependent on the (programming) language, locale, env (OS etc.).

Reply:
OK, I keep it like that.